packages/analytics/src/integrations/Integration.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-unused-vars */
+
+import type ConsentData from '../types/consentData.types';
+
+export interface IntegrationFactory {
+  new (
+    options: Record<string, unknown>,
+    loadData: Record<string, unknown>,
```

Comment:
loadData should be typed as the `EventData` type you have in `eventData.types`. However, there is one nuance here: The `type` and `event` properties contain the value `loadIntegration`. You might need to define additional types for those event cases, for example you might define a LoadIntegrationEventType that extends EventData and override the `type` and `event` properties with the type `loadIntegration` and use that type to type the loadData parameter..